src/reports_updater.rs

```diff
@@ -369,4 +437,59 @@ mod tests {
     let observed_tc_from_bytes = ObservedTcn::from_bytes(observed_tcn_as_bytes);
     assert_eq!(observed_tc_from_bytes, observed_tcn);
   }
+
+  #[test]
+  fn matching_benchmark() {
+
+    let verification_report_str = "D7Z8XrufMgfsFH3K5COnv17IFG2ahDb4VM/UMK/5y0+/OtUVVTh7sN0DQ5+R+ocecTilR+SIIpPHzujeJdJzugEAECcAFAEAmmq5XgAAAACaarleAAAAACEBo8p1WdGeXb5O5/3kN6x7GSylgiYGIGsABl3NrxhJu9XHwsN3f6yvRwUxs2fhP4oU5E3+JWabBP6v09pGV1xRCw==";
+    let verification_report_tcn: [u8; 16] = [24, 229, 125, 245, 98, 86, 219, 221, 172, 25, 232, 150, 206, 66, 164, 173]; // belongs to report
+    let verification_contact_time =  UnixTime { value: 1590528300 };
+    let verification_report = SignedReport::with_str(verification_report_str).unwrap();
+
+    let mut reports: Vec<SignedReport> = vec![0; 20].into_iter().map(|_| create_test_report()).collect();
+    reports.push(verification_report);
+
+    // let matcher = TcnMatcherStdThreadSpawn {}; // 20 -> 1s, 200 -> 16s, 1000 -> 84s, 10000 ->
+    let matcher = TcnMatcherRayon {}; // 20 -> 1s, 200 -> 7s, 1000 -> 87s, 10000 -> 927s
+
+    let tcns = vec![
+      ObservedTcn { tcn: TemporaryContactNumber([0; 16]), time: UnixTime { value: 1590528300 } },
+      ObservedTcn { tcn: TemporaryContactNumber(verification_report_tcn), time:  verification_contact_time.clone() },
+      ObservedTcn { tcn: TemporaryContactNumber([1; 16]), time: UnixTime { value: 1590528300 } },
+    ];
+
+    let matching_start_time = Instant::now();
+
+    let res = matcher.match_reports(tcns, reports);
```

Comment:
@i-schuetz  There seems to be some support for performance testing (unstable though): https://doc.rust-lang.org/unstable-book/library-features/test.html

Reply:
Ah cool! We can consider it for future benchmarking or refactorings. For now, given that we can tune easily the speed of this test by changing the number of reports, that it's an unstable feature and our code is still a bit "drafty" in general, I'd not prioritize it.